### PR TITLE
Implement Weasand

### PR DESCRIPTION
**Pull request by romanogit**

### CODE REVIEW

**Comment by jeremylarner** [comment]
Why do you need `ignoreWard`?

File: server/game/TerminalCondition.js [hunk 1]
```diff
@@ -26,7 +25,7 @@ class TerminalCondition {
     }
 
     checkCondition() {
-        return this.condition() && (!this.event || this.event.cancelled);
+        return this.condition() && !this.event;
```
Comment (jeremylarner) [inline server/game/TerminalCondition.js]: why are you removing this?  Can you test Weasand with an Armageddon Cloak?

File: server/game/cards/01-Core/TirelessCrocag.js [hunk 2]
```diff
@@ -6,19 +6,22 @@ class TirelessCrocag extends Card {
             condition: context => context.player.opponent.creaturesInPlay.length === 0,
```
Comment (jeremylarner) [inline server/game/cards/01-Core/TirelessCrocag.js]: If you've got terminal conditions to work, this play ability shouldn't be here

File: server/game/effectengine.js [hunk 3]
```diff
@@ -56,7 +56,8 @@ class EffectEngine {
     checkTerminalConditions() {
         let effectsToTrigger = this.terminalConditions.filter(effect => effect.checkCondition());
         if(effectsToTrigger.length > 0) {
-            this.game.openThenEventWindow(effectsToTrigger.map(effect => effect.getEvent()));
+            let events = effectsToTrigger.reduce((events, effect) => events.concat(effect.getEvent()), []);
```
Comment (jeremylarner) [inline server/game/effectengine.js]: why change it from a map to reduce?

File: server/game/TerminalCondition.js (hunk 1 above)
Comment (romanogit) [reply server/game/TerminalCondition.js]: Rolled back TerminalCondition to what it was and added test case to test Armageddon Cloak. It does not work. The creature does not leave play.

File: server/game/cards/01-Core/TirelessCrocag.js (hunk 2 above)
Comment (romanogit) [reply server/game/cards/01-Core/TirelessCrocag.js]: Correct. I will remove it.

File: server/game/effectengine.js (hunk 3 above)
Comment (romanogit) [reply server/game/effectengine.js]: I will rollback this change as well

**Pull request merged**